cadc-log/src/main/java/ca/nrc/cadc/log/ServletLogInfo.java

```diff
@@ -83,65 +86,38 @@
  */
 public class ServletLogInfo extends WebServiceLogInfo {
 
+    private static Logger log = Logger.getLogger(ServletLogInfo.class);
+
     /**
      * Servlet request constructor that automatically
      * determines the path.
      *
      * @param request The HTTP Request.
      */
     public ServletLogInfo(HttpServletRequest request) {
-        this(request, request.getPathInfo());
-    }
-
-    public ServletLogInfo(HttpServletRequest request, boolean pathIsJobID) {
-        this(request, request.getPathInfo(), true);
-    }
-
-    /**
-     * Servlet request constructor that takes a path
-     * override parameter.
-     *
-     * @param request The HTTP Request.
-     * @param path The path to log.
-     */
-    public ServletLogInfo(HttpServletRequest request, String path) {
-        this(request, path, false);
-    }
-
-    public ServletLogInfo(HttpServletRequest request, String path, String jobID) {
-        this(request, path, false);
-        this.jobID = jobID;
-    }
-
-    public ServletLogInfo(HttpServletRequest request, String path, boolean pathIsJobID) {
         super();
         this.method = request.getMethod().toUpperCase();
-        this.from = NetUtil.getClientIP(request);
-        if (pathIsJobID) {
-            this.jobID = parseJobID(path);
-        } else {
-            this.path = path;
+        this.ip = NetUtil.getClientIP(request);
+        String contextPath = request.getContextPath();
+        this.serviceName = this.parseServiceName(contextPath);
+        String servletPath = request.getServletPath();
+        if (servletPath == null) {
+            servletPath = "";
         }
-        Enumeration<String> paramNames = request.getParameterNames();
-        while (paramNames.hasMoreElements()) {
-            String nextName = paramNames.nextElement();
-            if (nextName.equalsIgnoreCase("runid")) {
-                runID = request.getParameter(nextName);
-                break;
-            }
+        String pathInfo = request.getPathInfo();
+        if (pathInfo == null) {
+            pathInfo = "";
```

Comment:
runID is no longer parsed out here.  How is it handled now?

Reply:
It's not. But runID is useful and should be set if available. Will check for it and set if available.